### ISSUE TITLE
Prepare cGAN Training

### ISSUE THREAD

**Issue by ckaarle** [ckaarle/ma#28]
- [x] store best verificator results
- [x] save checkpoint (incl. epoch and optimizer state) for both GANs
- [x] make parameters configurable
- [x] save samples images as cGAN performance evaluation
- [x] test SSIM
- [x] test all loss functions
- [x] figure out what to evaluate (z.B. mean distance original - generated image, all loss functions)
- [x] Training Setup with no code duplication between Repo and Notebook (python package)
